### ISSUE TITLE
Shared Memory Module: Implementation

### ISSUE THREAD

**Issue by ibrahim-1811** [HBRS-SDP/ss24-multi-robot-task-distribution#10]
- [x] Pipelines to get required data
- [ ] Design Database
- [x] Process the input data and store the required information
- [x] Pipelines for robot to access and update the shared memory

**Comment by nigitha03** [comment]
Database was not required as we are using in-memory storage. Will discuss with mentor to understand what data needs to be logged.

**Comment by ibrahim-1811** [comment]
yeah, but representation of data which is in shared memory was always requirement
